pkg/telemetry/platform.go

```diff
@@ -0,0 +1,99 @@
+package telemetry
+
+import (
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	openshiftIdentifier = "node.openshift.io/os_id"
+	k3sIdentifier       = "k3s"
+	awsIdentifier       = "aws"
+	gkeIdentifier       = "gce"
+	azureIdentifier     = "azure"
+	kindIdentifier      = "kind"
+	rancherIdentifier   = "cattle-system"
+)
+
+func CollectK8sPlatform(node v1.Node, namespaces v1.NamespaceList) string {
```

Comment:
For results, better to define a type and constants. For example:
```
type ClusterPlatform string

const (
	ClusterPlatformGKE       ClusterPlatform = "gke"
	ClusterPlatformEKS       ClusterPlatform = "eks"
	ClusterPlatformOpenShift ClusterPlatform = "openshift"
	ClusterPlatformUnknown   ClusterPlatform = "unknown"
)
```

Reply:
I made the changes but didn't choose to create the `type ClusterPlatform string` 

Keeping the return value as a `string`, I'm not too familiar with Golang still and couldn't find a way to return a ClusterPlatform without doing something like

`return string(clusterPlatformUnknown)` and that seemed a bit unnecessary. Is there a benefit of using ClusterPlatform that I am missing?